### DISCUSSION TITLE
How to connect carla client to the server

### DISCUSSION THREAD

**Discussion by vagaiprem** [carla-simulator/carla#6654] [General]
Hi Team 
      I am beginner for this Carla simulation. I launch Carla successfully but not able to connect Carla client in windows. Anyone tell me how to connect the Carla client to the server for windows?   plz help me.

**Comment by Najiya-08** [comment]
1.You need to install same version of carla in client machine 
2. One server machine:
./CarlaUE4.sh -quality-level=Low -carla-server --carla-port=2000
3. client machine
ssh -L 2000:localhost:2000 -L 2001:localhost:2001 -L 2002:localhost:2002 <IP address of server machine>
You can checkout this link as well : https://rocketloop.de/en/blog/carla-setup-remote-access/
4. After that run "python manual_control.py" in client machine

**Comment by vagaiprem** [comment]
ok tq so much Najiya. In single windows machine how we can execute this ?

**Comment by vagaiprem** [comment]
Najiya can u plz guide me and the solve the following problems.
1. windows carla version i have installed and working fine 
2. I want to know how to create carla client in same machine and execute different python scripts 
3.  plz tell me the step by step instructions
4.  i am waiting for your reply

**Comment by Najiya-08** [comment]
As I am working in ubuntu but the process is same in windows as well, in
same machine u don't need to create client separately. If your carla is
download then run the command for carla simulator in one terminal and run
python script in other terminal. It will start your clients. You can check
the working of carla in windows in YouTube. Tyere are multiple videos from
which you can get help

**Reply by vagaiprem** [reply]
Hi Najiya 
           When i am running python script in another terminal i am getting error as follows:
SystemError: type Boost.Python.enum has the Py_TPFLAGS_HAVE_GC flag but has no traverse function

can u help in this regard 

i am waiting for your response Njiya

**Comment by Najiya-08** [comment]


**Reply by vagaiprem** [reply]
Hi Najiya 
             can u able to solve my problem 
             carla Engine is running but if i run python script in another terminal i am getting this error

**Reply by vagaiprem** [reply]
i am using same windows machine for server as well as client

**Reply by Najiya-08** [reply]
Had you created a conda environment? As I had created a conda environment inside which I installed Carla and inside that environment only I am running all my code and command and its working fine, doesn't matter if I open multiple terminals.

**Reply by vagaiprem** [reply]
Hi Najiya:
I have installed directly in C drive . 
https://www.youtube.com/watch?v=lLkFA0fPrgs   - as per this video instruction i did everything but unable to run the python script.
if possible mail - technicalprem@gmail.com

**Reply by vagaiprem** [reply]
Hi Najiya 
                 I have created the environment in Conda and using VS code i can able to run the Carla Engine.
But when i am trying to run the python script it shows the following error:  If u knows kindly help me : 

Traceback (most recent call last):
  File "C:\prem\software\CARLA_0.9.14\WindowsNoEditor\PythonAPI\examples\manual_control.py", line 83, in <module>
    import carla
  File "C:\Users\Admin\anaconda3\envs\Carla_1\Lib\site-packages\carla\__init__.py", line 8, in <module>
    from .libcarla import *
SystemError: type Boost.Python.enum has the Py_TPFLAGS_HAVE_GC flag but has no traverse function
